### PR TITLE
Add core package test coverage (Phase 2)

### PR DESCRIPTION
**Pull request by stuckj**
## Summary
- Expand test coverage for core packages as part of Phase 2 of #45
- Added comprehensive error path and edge case tests

### Coverage Improvements
| Package | Before | After | Change |
|---------|--------|-------|--------|
| dedup | 70.2% | 79.6% | +9.4% |
| mkv | 28.9% | 45.5% | +16.6% |
| source | 22.7% | 25.9% | +3.2% |
| matcher | 6.2% | 9.5% | +3.3% |

### New Tests Added

**dedup/reader_test.go:**
- NewReader error paths (file not found, invalid magic, version errors)
- LoadSourceFiles with missing files
- ReadAt boundary conditions (beyond EOF, at exact end, empty buffer)
- VerifyIntegrity tests (valid file, corrupted index, invalid footer)
- Zero entries file handling
- Accessor method tests (OriginalSize, OriginalChecksum, SourceFiles, etc.)

**mkv/ebml_test.go:**
- Longer VINT values (5-8 bytes)
- Truncated data error handling
- ReadBinary, ReadInt tests
- Parser tests (NewParser error, empty file, accessors)
- Element ID verification
- Top-level element detection

**source/source_test.go:**
- DetectType with non-existent/empty directories
- NewIndexer error handling and window size clamping
- Index Close idempotency
- ReadRawDataAt/RawSlice/ReadESDataAt error paths
- ComputeHash tests

**matcher/matcher_test.go:**
- NewMatcher creation and fields
- SetNumWorkers with clamping
- Close idempotency
- Entry/Result/ProbeHash struct field tests
- ExtractProbeHashes edge cases

## Test plan
- [x] All new tests pass locally
- [x] No regressions in existing tests
- [x] CI passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/dedup/reader_test.go [hunk 1]
```diff
@@ -474,3 +474,484 @@ func TestEntryCount(t *testing.T) {
...
+
+	// Delta offset and size (right after header since no source files or entries)
+	deltaOffset := int64(HeaderSize)
+	deltaSize := int64(0)
+	binary.Write(f, binary.LittleEndian, deltaOffset)
+	binary.Write(f, binary.LittleEndian, deltaSize)
+
+	// Write footer (no index to checksum, no delta to checksum)
+	binary.Write(f, binary.LittleEndian, uint64(0xef46db3751d8e999)) // xxhash of empty
+	binary.Write(f, binary.LittleEndian, uint64(0xef46db3751d8e999)) // xxhash of empty
```
[the remark this answers is not in the document]
Comment (stuckj) [reply internal/dedup/reader_test.go]: Fixed - now computing the hash dynamically with `xxhash.Sum64(nil)`.

File: internal/source/source_test.go [hunk 2]
```diff
@@ -230,3 +230,264 @@ func TestIndex_Lookup(t *testing.T) {
...
+	}
+}
+
+func TestComputeHash_Empty(t *testing.T) {
+	// Empty data should produce a valid hash
+	hash := ComputeHash([]byte{})
+	if hash == 0 {
+		t.Error("ComputeHash() of empty data should not be zero")
+	}
+}
```
[the remark this answers is not in the document]
Comment (stuckj) [reply internal/source/source_test.go]: Fixed - refactored the test to check for consistency (same hash on multiple calls) rather than assuming the hash won't be zero. Also added a check that empty data produces a different hash than non-empty data.

File: internal/mkv/ebml_test.go [hunk 3]
```diff
@@ -415,3 +416,468 @@ func TestIsUnknownSize(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("Failed to create empty file: %v", err)
+	}
+	f.Close()
+
+	parser, err := NewParser(emptyFile)
+	if err != nil {
+		// Empty file might fail to open, that's ok
+		return
+	}
```
[the remark this answers is not in the document]
Comment (stuckj) [reply internal/mkv/ebml_test.go]: Fixed - improved the test to log when NewParser fails (expected behavior) and added a separate TestParser_EmptyContent test that creates a minimal file that may pass NewParser but fail during Parse().

File: internal/dedup/reader_test.go [hunk 4]
```diff
@@ -474,3 +475,485 @@ func TestEntryCount(t *testing.T) {
...
+	if info["entry_count"].(int) != 10 {
+		t.Errorf("info[entry_count] = %v, want 10", info["entry_count"])
+	}
+	if info["source_file_count"].(int) != 1 {
+		t.Errorf("info[source_file_count] = %v, want 1", info["source_file_count"])
+	}
+}
+
+// createTestDedupFileZeroEntries creates a valid dedup file with zero entries
+func createTestDedupFileZeroEntries(t *testing.T, tmpDir string) string {
```
[the remark this answers is not in the document]
Comment (stuckj) [reply internal/dedup/reader_test.go]: Fixed - introduced testDedupFileOptions struct and createTestDedupFileWithOptions helper. Now createTestDedupFile and createTestDedupFileZeroEntries both delegate to the shared helper with appropriate options, eliminating the duplication.
